mockall/tests/automock_extern_std.rs

```diff
@@ -0,0 +1,19 @@
+// vim: tw=80
+//! automocking a trait when std is only an extern crate (eg., as a testing
+//! support mod for a no_std library). This setup requires some extra "use"s
+//! to make, eg., Box, available.
+
+#![no_std]
+#[macro_use] extern crate std;
```

Comment:
Thanks for your review!

> Hey, this looks great! I have a question regarding `no_std`: is it compatible with the built in test support? That is, will there still be a test runner that executes the tests, or do you need to supply your own test runner?

Yes, everything is the same via cargo. Still do "cargo init --lib" and write tests etc. as is. Only difference is that "#![no_std]" is manually added to lib.rs to make it not link to the std runtime by default.

I never looked at it before under the hood so gave it a look via "cargo expand": the test binary needs "extern crate test", which, interesting enough, already depends on std internally: https://doc.rust-lang.org/src/test/lib.rs.html#63
This is of course understandable, because it doesn't make much sense to limit oneself to the no_std setup when writing tests.

> Is the `#[macro_use]` line still necessary in edition 2018?

Didn't notice this recommendation. Indeed I should import the macros just like the normal items instead of using this old construction "#[macro_use]".

As a side effect, a few places I need to fully qualify the "std::format!" inside the codebase. This is different from Box and Vec, where there's a common "use" block on all places where they are mentioned. "format!" seems more over multiple places.